src/network/connectionpool.py

```diff
@@ -127,32 +140,34 @@ def startUDPSocket(self, bind=None):
         self.udpSockets[udpSocket.listening.host] = udpSocket
 
     def loop(self):
-        # defaults to empty loop if outbound connections are maxed
+        """defaults to empty loop if outbound connections are maxed"""
+        # pylint: disable=too-many-statements,too-many-branches,too-many-nested-blocks
         spawnConnections = False
         acceptConnections = True
         if BMConfigParser().safeGetBoolean('bitmessagesettings', 'dontconnect'):
             acceptConnections = False
         elif BMConfigParser().safeGetBoolean('bitmessagesettings', 'sendoutgoingconnections'):
             spawnConnections = True
         if BMConfigParser().get('bitmessagesettings', 'socksproxytype')[0:5] == 'SOCKS' and \
-            (not BMConfigParser().getboolean('bitmessagesettings', 'sockslisten') and \
-            ".onion" not in BMConfigParser().get('bitmessagesettings', 'onionhostname')):
+            (not BMConfigParser().getboolean('bitmessagesettings', 'sockslisten') and
+             ".onion" not in BMConfigParser().get('bitmessagesettings', 'onionhostname')):
             acceptConnections = False
 
         if spawnConnections:
             if not knownnodes.knownNodesActual:
                 helper_bootstrap.dns()
             if not self.bootstrapped:
+                helper_bootstrap.dns()
```

Comment:
This may have changed in the meantime, please rebase so that I can see.

Reply:
Rebased.

Reply:
No, `helper_bootstrap.dns()` is above. That's enough.

Reply:
I'll accept it as it is now, it isn't a code quality issue, it can be changed in a separate PR.

Reply:
@PeterSurda any PR makes no sense if changes reverted soon after merge.
I spent a week to find a proper way to check knownnodes actuality. If someone thinks I did it wrong, he should create a separate issue, not revert my changes in unrelated PR.

Reply:
I don't have time to debug so let's go with what @g1itch said as he tested it. Please revert.

Reply:
So, I removed line 160 which I guess must have been added by git during the rebase. All other active lines are as they were. If there's something else I'm supposed to have reverted, let me know.